homeassistant/components/zwave_me/config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Config flow to configure ZWaveMe integration."""
+
+import logging
+
+from url_normalize import url_normalize
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from . import helpers
+from .const import CONF_TOKEN, CONF_URL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZWaveMeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """ZWaveMe integration config flow."""
+
+    def __init__(self):
+        """Initialize flow."""
+        self.url = None
+        self.token = None
+        self.uuid = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user or started with zeroconf."""
+        errors = {}
+        if self.url is None:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_URL): str,
+                    vol.Required(CONF_TOKEN): str,
+                }
+            )
+        else:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_TOKEN): str,
+                }
+            )
+
+        if user_input is not None:
+            if self.url is None:
+                self.url = user_input["url"]
+
+            self.token = user_input["token"]
+            if not self.url.startswith(("ws://", "wss://")):
+                self.url = f"ws://{self.url}"
+            self.url = url_normalize(self.url, default_scheme="ws")
+            if self.uuid is None:
+                self.uuid = await helpers.get_uuid(self.url, self.token)
+                if self.uuid is not None:
+                    await self.async_set_unique_id(f"{self.uuid}_{self.url}")
```

Comment:
Can we remove the url from the unique_id? The url is likely not stable if it's an ip address. The unique id needs to be both unique and stable.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements